### PR TITLE
Fix constness + reduce Boost dependency.

### PR DESCRIPTION
**Pull request by jmirabel**

### CODE REVIEW

File: include/hpp/fcl/profile.h [hunk 1]
```diff
@@ -170,6 +169,9 @@ class HPP_FCL_DLLAPI Profiler : private boost::noncopyable
       start();
   }
 
+  Profiler(const Profiler&) = delete;
```
Comment (jcarpent) [inline include/hpp/fcl/profile.h:172]: This is a C++11 feature!

**Review by jcarpent — CHANGES_REQUESTED** [review]
You are forcing the compilation against C++11, which is not something appealing at the current stage of the project, to just remove a minor dependency to Boost.

This is a bit too early I would say.

File: include/hpp/fcl/profile.h (hunk 1 above)
Comment (jmirabel) [reply include/hpp/fcl/profile.h]: You are correct. This was not my first intention. This file is anyway not used by anyone. I removed it.

**Comment by jmirabel** [comment]
@jcarpent any idea why windows CI fails ?

**Comment by jcarpent** [comment]
You should have a look to https://github.com/stack-of-tasks/pinocchio/commit/c0a3381de8681c4f38bde1a3dcd7894f300744e1#diff-290e375cf6344cd41694325176f37fab181cde59e19180067c1d37902af69b2c

**Comment by jmirabel** [comment]
@jcarpent if it's okay, can you modify your review ? Now the code is C++98 compliant.

**Review by jcarpent — APPROVED** [review]

**Pull request merged**